src/appleseed.studio/mainwindow/mainwindow.cpp

```diff
@@ -902,6 +936,13 @@ bool MainWindow::can_close_project()
     return false;
 }
 
+bool MainWindow::can_close_project()
+{
+    return !m_project_manager.is_project_loading() && // Project is not being loaded: no problem
```

Comment:
Rewrite as a series of `if` with early exits as it used to be done. This makes the logic much clearer.